### PR TITLE
Wrap up all the refection and linq expression usage in try-catch 

### PR DESCRIPTION
**Pull request by Cosifne**
A regression is introduced in 17.9 P2 because of the AI rename.
We use reflection & linq expression to build the interface wrapper, however, this path is fragile, if editor folks change anything, it will end up breaks us, cause the user can't use the normal inline rename.

This PR wraps all these calls with try-catch, and only build `SmartRenameViewModel` when things are correct. In case there are errors, user would still fall back to the normal UI

### CODE REVIEW

**Review by sharwell — CHANGES_REQUESTED** [review]
marking to give time for review

File: src/EditorFeatures/Core.Wpf/Lightup/ISmartRenameSessionWrapper.cs [hunk 1]
```diff
@@ -17,83 +19,94 @@ namespace Microsoft.CodeAnalysis.EditorFeatures.Lightup;
...
-    private static readonly Func<object, bool> s_isInProgressAccessor;
-    private static readonly Func<object, string> s_statusMessageAccessor;
-    private static readonly Func<object, bool> s_statusMessageVisibilityAccessor;
-    private static readonly Func<object, IReadOnlyList<string>> s_suggestedNamesAccessor;
+    private static readonly Func<object, bool>? s_isAvailableAccessor;
+    private static readonly Func<object, bool>? s_hasSuggestionsAccessor;
+    private static readonly Func<object, bool>? s_isInProgressAccessor;
+    private static readonly Func<object, string>? s_statusMessageAccessor;
+    private static readonly Func<object, bool>? s_statusMessageVisibilityAccessor;
+    private static readonly Func<object, IReadOnlyList<string>>? s_suggestedNamesAccessor;
```
Comment (sharwell) [inline src/EditorFeatures/Core.Wpf/Lightup/ISmartRenameSessionWrapper.cs:29]: None of these should be nullable. If the member cannot be bound, an implementation returning a default value should be used instead.

**Comment by Cosifne** [comment]
I have an offline discussion with Sam. Close this now

**Comment by CyrusNajmabadi** [comment]
What's the plan on making us resilient here?  Thanks!

**Comment by Cosifne** [comment]
> What's the plan on making us resilient here? Thanks!

@CyrusNajmabadi 
See the explanation here https://github.com/dotnet/roslyn/issues/71410#issuecomment-1877765499

**Comment by sharwell** [comment]
@CyrusNajmabadi A lot of the issue here is I didn't have access to the internal feature for testing at the time I created the lightup layer, so it was failing early.

**Pull request closed without merging**